python/llm/src/ipex_llm/transformers/models/chatglm4.py

```diff
@@ -46,10 +46,11 @@ def chatglm4_model_forward(
     use_cache = use_cache if use_cache is not None else self.config.use_cache
     return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
-    batch_size, seq_length = input_ids.shape
-
     if inputs_embeds is None:
+        batch_size, seq_length = input_ids.shape
         inputs_embeds = self.embedding(input_ids)
+    else:
+        batch_size, seq_length, _ = inputs_embeds.shape
```

Comment:
If attention_mask is given a not-None and not-All value, `input_ids` is needed in line 58 `self.get_masks()` and it will raise error if `input_ids` is still None. Maybe add an empty tensor here? 
```python
input_ids = torch.empty((batch_size, seq_length), device=inputs_embeds.device)
```

Reply:
Relative code in modeling_chatglm:
```python
    def get_masks(self, input_ids, past_key_values, padding_mask=None):
        batch_size, seq_length = input_ids.shape
```

Reply:
> If attention_mask is given a not-None and not-All value, `input_ids` is needed in line 58 `self.get_masks()` and it will raise error if `input_ids` is still None. Maybe add an empty tensor here?

Have updated in `chatglm2.py` and `chatglm4.py` :)